app/controllers/csv_preview_controller.rb

```diff
@@ -30,18 +30,7 @@ def attachment_visible?
   end
 
   def fail
-    if (edition = attachment_visibility.unpublished_edition)
-      redirect_to edition.unpublishing.document_path
-    elsif (replacement = attachment_data.replaced_by)
-      expires_headers
-      redirect_to replacement.url, status: 301
-    elsif image? upload_path
-      redirect_to view_context.path_to_image('thumbnail-placeholder.png')
-    elsif incoming_upload_exists? upload_path
-      redirect_to_placeholder
-    else
-      render plain: "Not found", status: :not_found
-    end
```

Comment:
It's not obvious to me that it makes sense to remove this code. Although there aren't tests exercising these paths, it seems quite likely that they're actually being used. I think we should probably add the missing tests rather than remove the functionality.